### PR TITLE
[DPS-1303] Rating feature

### PR DESCRIPTION
**Pull request by ale-kacperczyk**
A new feature letting us gather rating & feedback on our documentation.

Feature branch:
https://developers.labs.livechatinc.com/docs/feature/dps-1303/

Design:
https://www.figma.com/file/wVsSvN52NqhzlBdjsi0Vnx/Documentation-Rating?node-id=0%3A1

Jira:
https://livechatinc.atlassian.net/browse/DPS-1303

### CODE REVIEW

File: src/components/core/Rating/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <Wrapper margin={margin}>
+      <Label margin={margin}>{label}</Label>
+      {RATES.map((r, i) => {
+        const isHover = hoverStar > i - 1;
+        const isSelected = selectedStar > i - 1;
+        const isRated = selectedStar === i;
+
+        return (
+          <Star
+            key={r}
```
Comment (jakubsikora) [inline src/components/core/Rating/index.js]: I'm not a big fan of minimising variables name when you don't need it. It's not good for code readability (maybe not in this case that much). JS minifier will handle that later.

```suggestion
      {RATES.map((rate, i) => {
        const isHover = hoverStar > i - 1;
        const isSelected = selectedStar > i - 1;
        const isRated = selectedStar === i;

        return (
          <Star
            key={rate}
```

File: src/hooks/useRating.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    rating => rating.pathname === window.location.pathname
+  );
+
+  const [selectedStar, setSelectedStar] = useState(
+    currentRating ? currentRating.rating : -1
+  );
+
+  const saveRating = index => {
+    const newRatings = ratings.filter(
+      r => r.pathname !== window.location.pathname
```
Comment (jakubsikora) [inline src/hooks/useRating.js]:
```suggestion
      rating => rating.pathname !== window.location.pathname
```

File: src/templates/page.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React, { useState, useEffect, useLayoutEffect } from "react";
 import { graphql, Link, navigate } from "gatsby";
 import { MDXRenderer } from "gatsby-plugin-mdx";
 import { MDXProvider } from "@mdx-js/react";
+import * as DesignSystem from "@livechat/design-system";
```
Comment (jakubsikora) [inline src/templates/page.js]: Do we need that import ?

**Comment by dominikslawkowski** [comment]
🌀waiting for @macpankiewicz design approve

File: src/templates/page.js [hunk 4]
```diff
@@ -156,45 +167,50 @@ export default ({ data: { mdx, allMdx } }) => {
     items: versions
   };
 
+  console.log("customSlug", customSlug);
```
Comment (SamborGornicz) [inline src/templates/page.js]: ;)

Comment (dominikslawkowski) [reply src/templates/page.js]: just testing on feature branch (changing the URL structure affects functionality)

Comment (dominikslawkowski) [reply src/templates/page.js]: ok I will use gatsby serve

File: src/templates/page.js [hunk 5]
```diff
@@ -18,26 +19,36 @@ import {
...
-  ...DesignSystem,
   ...CodeBlocks,
   ...Headings,
   RichMessagePreview,
   Link
 };
 
+const StyledRating = styled(Rating)`
+  width: 270px;
+  margin: 50px 0 0 calc(50% - 135px);
```
Comment (jakubsikora) [inline src/templates/page.js]: You can just wrap rating component with some div position flex, justify-content: center and you have your rating centered without margin calculation

File: src/templates/page.js [hunk 6]
```diff
@@ -156,45 +167,50 @@ export default ({ data: { mdx, allMdx } }) => {
     items: versions
   };
 
+  console.log("customSlug", slug);
```
Comment (jakubsikora) [inline src/templates/page.js]: :)

**Pull request merged**